### PR TITLE
Discuss inline functions and variables

### PR DESCRIPTION
**Pull request by bernhardmgruber**
Fixes: #175

### CODE REVIEW

File: talk/basicconcepts/inline.tex [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    \end{cppcode*}
+  \end{block}
+\end{frame}
+
+\begin{frame}[fragile]
+  \frametitlecpp[98]{Inline keyword}
+  \begin{block}{Inline functions nowadays}
+    \begin{itemize}
+      \item compilers can judge far better when to inline or not
+      \item side effect became primary purpose: to put functions into headers
```
Comment (sponce) [inline talk/basicconcepts/inline.tex:28]:
```suggestion
      \item compilers can judge far better when to inline or not
        \begin{itemize}
        \item thus primary purpose is gone
        \end{itemize}
      \item put functions into headers became main purpose
```

File: talk/basicconcepts/inline.tex [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  \end{block}
+\end{frame}
+
+\begin{frame}[fragile]
+  \frametitlecpp[17]{Inline keyword}
+  \begin{block}{Inline variables}
+    \begin{itemize}
+      \item we can also mark a global (or \mintinline{cpp}{static} member) variable as inline
+      \item same side effect, linker merges all occurrences into one
+      \item useful for defining global variables/constants in headers
```
Comment (sponce) [inline talk/basicconcepts/inline.tex:45]:
```suggestion
      \item allows to define global variables/constants in headers
```

Comment (bernhardmgruber) [reply talk/basicconcepts/inline.tex:45]: Yeah, let's not endorse that too much :D

File: talk/basicconcepts/inline.tex [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  \frametitlecpp[98]{Inline keyword}
+  \begin{block}{Inline functions originally}
+    \begin{itemize}
+      \item can be applied to a function to tell the compiler to inline it
+      \item just a hint, compiler can still choose to inline/not inline.
+      \item effect: replace a function call by the function's content, like a macro
+      \item avoids function call overhead, but may increase executable size if inlined in many places
+      \item side effect: an inline function definition can be put into header files, the linker reduces the duplicated functions into one
+    \end{itemize}
+  \end{block}
```
[the remark this answers is not in the document]
Comment (bernhardmgruber) [reply talk/basicconcepts/inline.tex]: Thx, I applied the suggested changes.

File: talk/basicconcepts/inline.tex (hunk 1 above)
Comment (bernhardmgruber) [reply talk/basicconcepts/inline.tex]: Thx, I applied the suggested changes.

**Comment by sponce** [comment]
Oh I did not even notice the bottom box ! It was not appearing at all before my changes :-D But is very welcome, I had it in mind when reading the slide actually ! What about simply have the code in `\small` font ? Most of the code is in small and it's still very readable.

**Comment by bernhardmgruber** [comment]
I added the box after your review, because I figured that I should point this out more clearly.

**Comment by bernhardmgruber** [comment]
Ok the `\small` worked:
image

**Pull request merged**